### PR TITLE
Add execution-unit to glossary.md

### LINKED ISSUE #1511: Add execution-unit to the glossary.
In the context section we use "execution unit" to talk about threads/coroutines/fibers/etc. We should add it to the glossary (or alternatively use a better term).

### PR DESCRIPTION
**Pull request by svrnm**
Fixes #1511 ; since I was looking into adding the term service already, I thought I could also give this one a try. Hope it helps:)

## Changes

- Update glossary to contain a definition of the term execution unit
- Link back from opentracing.md and context.md where this term is used

### CODE REVIEW

File: specification/compatibility/opentracing.md [hunk 1]
```diff
@@ -126,7 +126,7 @@ MUST be done by the Shim layer in order to retain these invariants.
 
 Because of the previous requirement, a given OpenTelemetry `Span`
 MUST be associated with ONE AND ONLY ONE `SpanContext` Shim object at all times
-for ALL execution units, in order to keep any linked `Baggage` consistent
+for ALL [execution units](../glossary.md#execution-unit), in order to keep any linked `Baggage` consistent
```
Comment (yurishkuro) [inline specification/compatibility/opentracing.md:129]: @carlosalberto I think this section is overly restrictive, which may cause difficulties in implementation. For example:

> OpenTracing `SpanContext`, just as OpenTelemetry `SpanContext`, MUST be immutable

I don't think this was an explicitly stated requirement in OT, because SpanContext was an empty interface, thus there was no way to distinguish mutable from immutable. However, It was allowed that `span.context().baggageItems()` could return different results at different times. Given that, the next requirement:

> a given OpenTelemetry `Span` MUST be associated with ONE AND ONLY ONE `SpanContext` Shim object at all times

also seems somewhat problematic (it's actually hard to follow why this is written this way). In contrast, in Jaeger SDKs `span.context()` would return an immutable object, but could return a different object (with different baggage) on another invocation of `span.context()`. The context was immutable only from the POV of public interface of jaeger.SpanContext - internally the object could still be mutable e.g. for tracking deferred sampling.

So in summary, this seems to be dictating a lot of restrictions which were not present in OT. It's fine if you think this restrictions actually help with the shim implementation.

**Review by yurishkuro — APPROVED** [review]
I'm fine with the PR itself, but have concerns with the existing section that uses the term.

File: specification/compatibility/opentracing.md (hunk 1 above)
Comment (carlosalberto) [reply specification/compatibility/opentracing.md:129]: @yurishkuro We will actually overhaul completely this entire section based on the discussion in #2137 -- that is, we will NOT have this requirement, and we will do as Jaeger does (which is what OT says as well):

> span.context() would return an immutable object, but could return a different object (with different baggage) on another invocation of span.context()

The paragraph above was, yes, an intent to help with the Shim layer, but as we will be taking a simpler, better, correct route, we don't need the text above.

**Review by carlosalberto — APPROVED** [review]

File: specification/compatibility/opentracing.md (hunk 1 above)
Comment (svrnm) [reply specification/compatibility/opentracing.md:129]: Looks like this is not part of this PR and is covered by another issue, so I can resolve this conversation?
